internal/exec/stages/files/filesystemEntries.go

```diff
@@ -16,16 +16,90 @@ package files
 
 import (
 	"fmt"
+	"io/ioutil"
 	"os"
+	"os/exec"
 	"path/filepath"
 	"sort"
 	"strings"
 
 	"github.com/coreos/ignition/v2/config/v3_2_experimental/types"
+	"github.com/coreos/ignition/v2/internal/distro"
 	"github.com/coreos/ignition/v2/internal/exec/util"
 	"github.com/coreos/ignition/v2/internal/log"
+
+	"github.com/vincent-petithory/dataurl"
 )
 
+// createCrypttabEntries creates entries inside of /etc/crypttab for LUKS volumes,
+// as well as copying keyfiles to the sysroot.
+func (s *stage) createCrypttabEntries(config types.Config) error {
+	if len(config.Storage.Luks) == 0 {
+		return nil
+	}
+
+	s.Logger.PushPrefix("createCrypttabEntries")
+	defer s.Logger.PopPrefix()
+
+	crypttab := fileEntry{
+		types.Node{
+			Path: "/sysroot/etc/crypttab",
+		},
+		types.FileEmbedded1{},
+	}
+	keyfiles := []fileEntry{}
+	for _, luks := range config.Storage.Luks {
+		out, err := exec.Command(distro.CryptsetupCmd(), "luksUUID", luks.Device).CombinedOutput()
+		if err != nil {
+			return fmt.Errorf("gathering luks uuid: %s: %v", out, err)
+		}
+		uuid := strings.TrimSpace(string(out))
+		netdev := ""
+		if len(luks.Clevis.Tang) > 0 {
```

Comment:
```suggestion
		if luks.Clevis != nil && len(luks.Clevis.Tang) > 0 {
```

This will `SIGSEGV` otherwise if `luks.Clevis` is `nil`.